q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,312 @@
+        else:
+            raise ValidationError(variable, rule_value)
+
+    elif rule == 'normalization':
+        if check_normalization(rule_value):
+            rules['edits'][rule] = rule_value
+        else:
+            raise NormalizationError(variable, rule_value)
+
+    elif rule == 'blank':
+        if check_str(rule_value):
+            rules['allowed'][rule] = rule_value
+        else:
+            raise BlankError(variable, rule_value)
+
+    elif rule == 'missing':
+        if check_str(rule_value):
+            rules['allowed'][rule] = rule_value
+        else:
+            raise MissingError(variable, rule_value)
+
+    elif rule == 'format':
+        if check_str(rule_value):
+            rules['format'] = rule_value
+        else:
+            raise FormatError(variable, rule_value)
+
+
+def check_expected(rule_value):
+    """Check that the user-defined value for the
+    rule "expected" is correctly formatted.
+
+    expected must be a list
+        Apply rule of allowing only expected values:
+        These do allow the missing and blanks.
+                    expected:
+                        - str
+                        - list
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "expected".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+
+    """
+    # check if the rule value is indeed a list (it must be the case).
+    if isinstance(rule_value, list):
+        # checks that all the items in the expected value list are strings.
+        str_items = [x for x in rule_value if isinstance(x, str)]
+        if len(str_items) == len(rule_value):
+            return True
+    return False
+
+
+def check_ontology(rule_value):
+    """Check that the user-defined value for the
+    rule "ontology" is correctly formatted.
+
+    ontology must be a str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    ontology: Gazetteer
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "ontology".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, str):
+        if rule_value in ['Gazetteer ontology']:
+            return True
+    return False
+
+
+def check_remap(rule_value):
+    """Check that the user-defined value for the
+    rule "remap" is correctly formatted.
+
+    remap must be a dict
+        Apply the rule of remapping variables values: replace
+        all instances of a given value by another one value.
+                    remap:
+                      str: str
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "remap".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+
+    """
+    if isinstance(rule_value, dict):
+        str_values = [x for x, y in rule_value.items() if isinstance(y, str)]
```

Comment:
I've re-thinked and since pandas can remap both numerics and chars by anything, then there should not be a restriction to string only

This will be pushed asap:

```suggestion
        str_values = [x for x, y in rule_value.items() if isinstance(x, (str, int, float)) or isinstance(y, (str, int, float))]
```